📝💄 Split the "process" change log entry to 3

webknjaz opened the pull request:
Previously, the change log had a section called "Process". And some of
the incoming pull requests would add change notes there when they
couldn't be put into other categories. Using it like that is not a
good idea. This patch rethinks the categorization and the audiences of
a few common change types and defines respective categories that
replace "Process".

The new categories are:

  * `packaging` -- news for downstream re-packagers

  * `contrib` -- news for regular project participants

  * `misc` -- public change log for things that don't fit anywhere but
    are useful

Resolves https://github.com/pypa/pip/issues/12555

Change note demo: https://pip--12853.org.readthedocs.build/en/12853/news/#contributor-facing-changes

webknjaz committed:
🤖 Reflow Towncrier categories list in config

This is a preparatory step for further adding of new changelog
fragment types.

webknjaz committed:
💅 Blame-ignore 3303cd1ce

Said commit is a formatting change.

webknjaz committed:
📝💄 Split the "process" change log entry to 3

Previously, the change log had a section called "Process". And some of
the incoming pull requests would add change notes there when they
couldn't be put into other categories. Using it like that is not a
good idea. This patch rethinks the categorization and the audiences of
a few common change types and defines respective categories that
replace "Process".

The new categories are:

  * `packaging` -- news for downstream re-packagers

  * `contrib` -- news for regular project participants

  * `misc` -- public change log for things that don't fit anywhere but
    are useful

Resolves #12555

sbidoul reviewed (approved):
Looks good, thanks! This PR is a candidate for a `contrib` news, right?

webknjaz commented:
> Looks good, thanks! This PR is a candidate for a `contrib` news, right?

@sbidoul yes! I adjusted a few things to make the linters pass and added that fragment. I was in a bit of a hurry to get off the train on the way home from EuroPython, and so I missed this. It should be good now.

N.B. I also noticed that despite the contrib guide saying that `trivial` fragments are to be empty, I've found a few with text. I didn't change them, since that's out of the scope, but we should probably lint this case and prevent it from happening. I filed https://github.com/sanitizers/chronographer-github-app/issues/39 to implement the check in the app at some point, but I'll try to also add a pre-commit based implementation for now.

webknjaz commented on docs/html/development/contributing.rst:
```diff
@@ -110,8 +110,8 @@ public is concerned, typo fixes, white space modification, etc. To mark a PR
 POSIX like operating system, one can be added by running
-``touch news/$(uuidgen).trivial.rst``. On Windows, the same result can be
-achieved in Powershell using ``New-Item "news/$([guid]::NewGuid()).trivial.rst"``.
+``touch news/+$(uuidgen).trivial.rst``. On Windows, the same result can be
```
@sbidoul FYI, fragments with the leading `+` are called “orphan” within Towncrier and so it's usually good to follow their conventions for this random stuff that might not need to link issue/PR numbers.

sbidoul replied:
TIL. Thanks!

webknjaz commented:
@pradyunsg ping

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -65,9 +65,45 @@ repos:
+            |contrib
+            |misc
+            |trivial
+          )(\.\d+)?(\.rst)?
+          |[0-9a-zA-Z-_]+\.vendor(\.\d+)?(\.rst)?  # special-case vendoring
```
```suggestion
          )(\.\d+)?\.rst
          |[0-9a-zA-Z-_]+\.vendor(\.\d+)?\.rst  # special-case vendoring
```

webknjaz commented:
@sbidoul so last time I discussed this with Pradyun in the summer, he wasn't happy about this change. Is there anything you'd like to salvage from this patch, or should we just close it as rejected?

sbidoul commented:
@webknjaz personally I'm fine with this PR, but I'm also fine with the status quo. So if Praduyn has concern, maybe give him time to chime in here?

webknjaz commented:
@sbidoul we talked privately back then and he said that it's not a good idea to put contrib notes there IIRC. But okay, let's wait. I just wanted to make sure this isn't cluttering the tracker for no reason...

ichard26 commented:
> so last time I discussed this with Pradyun in the summer, he wasn't happy about this change. Is there anything you'd like to salvage from this patch, or should we just close it as rejected?

I'll consider this as rejected. We don't use the process fragment much, so like @sbidoul, the statuo quo is fine IMO.

Thanks for the effort, either way.
